### PR TITLE
Re-orgnize the controllers package

### PR DESCRIPTION
**Pull request by LinuxSuRen**
### What type of PR is this?

/kind chore

### What this PR does / why we need it:
It can help contributors to understand all the controllers.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:

I deleted an useless CRD: `fake`

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

File: pkg/api/devops/v1alpha3/register.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+func init() {
```
Comment (JohnNiang) [inline pkg/api/devops/v1alpha3/register.go:19]: I don't think we need to register schemes here. Please see 

https://github.com/kubesphere/ks-devops/blob/c6d6f2a3b4abad901727ca468175cb4a3441de58/pkg/api/devops/v1alpha3/pipeline_types.go#L259-L261

https://github.com/kubesphere/ks-devops/blob/c6d6f2a3b4abad901727ca468175cb4a3441de58/pkg/api/devops/v1alpha3/pipelinerun_types.go#L303-L305

https://github.com/kubesphere/ks-devops/blob/c6d6f2a3b4abad901727ca468175cb4a3441de58/pkg/api/devops/v1alpha3/devopsproject_types.go#L71-L73

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: pkg/api/devops/v1alpha3/register.go (hunk 1 above)
Comment (LinuxSuRen) [reply pkg/api/devops/v1alpha3/register.go:19]: I have removed all the `Registers` from each go source file. It is not a good practice to have many `init` functions.

Comment (JohnNiang) [reply pkg/api/devops/v1alpha3/register.go:19]: Those init functions are generated by `kubebuilder` when creating that API. You can see the official sample layout: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cronjob-tutorial/testdata/project/api/v1.

BTW, is there any reference to show many `init` functions are not a good practice? You can check this: https://go.dev/doc/effective_go#init as well.

Comment (LinuxSuRen) [reply pkg/api/devops/v1alpha3/register.go:19]: It's more clear if we put all register together. And it's not necessary to have too many `init` functions that have the same purpose. Of course, this is not a big deal if you insist on negative feedback.

**Review by JohnNiang — APPROVED** [review]
/lgtm
/label tide/merge-method-squash

**Pull request merged**